### ISSUE TITLE
Web: onClick fails with ClassCastException on Safari Mobile

### ISSUE THREAD

**Issue by hannesstruss** [JetBrains/compose-jb#1053]
In this app:

```kotlin
fun main() {
  renderComposable(rootElementId = "root") {
    Button(attrs = {
      onClick { console.log("Hello") }
    }) { 
      Text("Test")
    }
  }
}
```

when clicking the button on Safari Mobile (iOS), a `ClassCastException` will be thrown while trying to convert the native event to `SyntheticMouseEvent`. Apparently on iOS devices the native event doesn't have a `movementX` value which seems to be required by `SyntheticMouseEvent`: https://github.com/JetBrains/compose-jb/blob/master/web/core/src/jsMain/kotlin/org/jetbrains/compose/web/events/SyntheticMouseEvent.kt#L26-L27

Using `attrs = { addEventListener("click") { ... } }` works.

Tested in `0.5.0-build270` and `1.0.0-alpha1-rc3`.

**Comment by CarsonRedeye** [comment]
Thanks for the workaround! I was stuck 😅

**Comment by MrPowerGamerBR** [comment]
Here's a small function that allows you to use the workaround just like how you would use the `onClick` function 😊

```kotlin
// Issue: https://github.com/JetBrains/compose-jb/issues/1053
fun AttrsBuilder<*>.onClickSafariWorkaround(callback: () -> (Unit)) {
    addEventListener("click") {
        callback.invoke()
    }
}
```

**Comment by ccjernigan** [comment]
This can be reproduced easily with one of the compose sample pages

Steps to reproduce
1. Visit https://compose-web.ui.pages.jetbrains.team/ on an iOS or iPadOS device (or simulator)
2. Tap the "Hello" button next to "Result"

Results:
Expected: Button changes text randomly.
Actual: Button does nothing

**Comment by eymar** [comment]
was fixed in https://github.com/JetBrains/compose-jb/pull/1288. I'll close this ticket as soon as the fix is released.

**Comment by eymar** [comment]
Closing this issue as it's fixed in 1.0.0-beta4

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-5304">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
